velox/exec/tests/AssignUniqueIdTest.cpp

```diff
@@ -92,3 +92,26 @@ TEST_F(AssignUniqueIdTest, multiThread) {
     verifyUniqueId(plan, input, 8);
   }
 }
+
+TEST_F(AssignUniqueIdTest, maxRowIdLimit) {
+  auto input = {
+      makeRowVector({makeFlatVector<bool>(1, [](auto row) { return row; })})};
+
+  auto plan = PlanBuilder().values(input).assignUniqueId().planNode();
+  // Increase the counter to kMaxRowId
+  std::dynamic_pointer_cast<core::AssignUniqueIdNode>(plan)
+      ->uniqueIdCounter()
+      ->fetch_add(1L << 40);
+
+  EXPECT_THROW(verifyUniqueId(plan, input), VeloxRuntimeError);
```

Comment:
Did this test hang indefinitely before the fix?

Reply:
yes